graphein/molecule/config.py

```diff
@@ -0,0 +1,117 @@
+"""Base Config object for use with Molecule Graph Construction."""
+# Graphein
+# Author: Arian Jamasb <arian@jamasb.io>
+# License: MIT
+# Project Website: https://github.com/a-r-j/graphein
+# Code Repository: https://github.com/a-r-j/graphein
+from __future__ import annotations
+
+from functools import partial
+from pathlib import Path
+from typing import Any, Callable, List, Optional, Union
+
+from deepdiff import DeepDiff
+from deepdiff.operator import BaseOperator
+from pydantic import BaseModel
+from typing_extensions import Literal
+
+from graphein.molecule.edges.atomic import add_atom_bonds
+
+from graphein.molecule.edges.distance import add_k_nn_edges, add_distance_threshold, add_fully_connected_edges
+
+from graphein.molecule.features.nodes.atom_type import atom_type_one_hot
+
+GraphAtoms = Literal[
+    "C",
+    "H",
+    "O",
+    "N",
+    "F",
+    "P",
+    "S",
+    "Cl",
+    "Br",
+    "I",
+    "B",
+]
+"""Allowable atom types for nodes in the graph."""
+
+def partial_functions_equal(func1: partial, func2: partial) -> bool:
```

Comment:
There's a lot of code duplication here with the protein module. I think I should refactor the shared functions into a utils submodule to avoid this.

Reply:
sounds great!